src/daily-dropoffs-report.tsx

```diff
@@ -0,0 +1,392 @@
+          steps.set(key, (row[key] || 0) + oldCount);
+        });
+      });
+
+      const { issuer, friendly_name: friendlyName, iaa, agency, start, finish } = bin[0];
+
+      return {
+        issuer,
+        friendly_name: friendlyName,
+        iaa,
+        agency,
+        start,
+        finish,
+        ...Object.fromEntries(steps),
+      } as DailyDropoffsRow;
+    });
+}
+
+function tabulate({
+  rows: results,
+  filterAgency,
+  issuerColor,
+}: {
+  rows?: DailyDropoffsRow[];
+  filterAgency?: string;
+  issuerColor: (issuer: string) => string;
+}): TableData {
+  const filteredRows = (results || []).filter((d) => !filterAgency || d.agency === filterAgency);
+
+  const header = [
+    "Agency",
+    "App",
+    ...STEPS.map(({ title }, idx) => <th colSpan={idx === 0 ? 1 : 2}>{title}</th>),
+  ];
+
+  const color = scaleLinear()
+    .domain([1, 0])
+    .range([
+      // scaleLinear can interpolate string colors, but the 3rd party type annotations don't know that yet
+      "steelblue" as unknown as number,
+      "white" as unknown as number,
+    ]);
+
+  const body = filteredRows.map((row) => {
+    const { agency, issuer, friendly_name: friendlyName } = row;
+
+    return [
+      agency,
+      <span title={issuer}>
+        <span style={`color: ${issuerColor(issuer)}`}>⬤ </span>
+        {friendlyName}
+      </span>,
+      ...toStepCounts(row).flatMap(({ count, percentOfFirst }, idx) => {
+        const backgroundColor = `background-color: ${color(percentOfFirst)};`;
+        const cells = [
+          <td className="table-number text-tabular text-right" style={backgroundColor}>
+            {formatWithCommas(count)}
+          </td>,
+        ];
+
+        if (idx > 0) {
+          cells.push(
+            <td className="table-number text-tabular text-right" style={backgroundColor}>
+              {formatAsPercent(percentOfFirst)}
+            </td>
+          );
+        }
+
+        return cells;
+      }),
+    ];
+  });
+
+  return {
+    header,
+    body,
+  };
+}
+
+function loadData(
+  start: Date,
+  finish: Date,
+  env: string,
+  fetch = window.fetch
+): Promise<DailyDropoffsRow[]> {
+  return Promise.all(
+    utcDays(start, finish, 1).map((date) => {
+      const path = reportPath({ reportName: "daily-dropoffs-report", date, env, extension: "csv" });
+      return fetch(path).then((response) => response.text());
+    })
+  ).then((reports) => aggregate(reports.flatMap((r) => process(r))));
+}
+
+function Axis({
+  axis,
+  transform,
+  rotateLabels,
+  className,
+}: {
+  axis: D3Axis<NumberValue> | D3Axis<string>;
+  transform: string;
+  rotateLabels?: boolean;
+  className?: string;
+}): VNode {
+  const ref = useRef(null as SVGGElement | null);
+
+  useEffect(() => {
+    if (ref.current) {
+      select(ref.current).call(axis).classed("rotate-labels", !!rotateLabels);
+    }
+  });
+
+  return <g ref={ref} className={className} transform={transform} />;
+}
+
+function LineChart({
```

Comment:
Can we pull out some of these components to separate files?

Reply:
So this LineChart is very tied to the structure of this particular report, so I'd pull it out as DropOffReportLineChart ?

Reply:
Ok so... this had a lot of circular dependencies... may have to separate this report's data model out into its own file so it can be shared.. will take a look tomorrow

Reply:
I have some ideas for breaking up the current report classes into separate data model vs rendering components... going to do that as a separate PR to minimize the noise

Reply:
Ah, yeah, the original comment was more in the spirit of "if this is just a generic component to render a line chart", toward an ideal of one-component-per-file. And if it's not a generic component, could it be?

Reply:
If Plot's line chart supported ordinal scales like this we could use that instead.... if we add a second line chart like this I could definitely make it more generic, but since it's only one of one, I think it works?